src/common/operator/cast_operators.cpp

```diff
@@ -820,4 +820,40 @@ template <> timestamp_t CastToTimestamp::Operation(string_t input) {
 	return Timestamp::FromString(input.GetData());
 }
 
+//===--------------------------------------------------------------------===//
+// Cast From Blob
+//===--------------------------------------------------------------------===//
+template <> string_t CastFromBlob::Operation(string_t input, Vector &vector) {
+	idx_t input_size = input.GetSize();
+	auto input_data = input.GetData();
+	// Check if the input is already a hexa string, i.e., avoiding a double convertion
+	// Returns just a copy of the input
+	if(input_size >= 2 && input_data[0] == '\\' && input_data[1] == 'x') {
```

Comment:
This is a bit hacky, the double conversion should be avoided at a higher level by keeping track of the SQL types. What if you have a blob that starts with the bytes "\x" followed by illegal UTF8 characters?